### PR TITLE
[AI] AI stream + tools + tests

### PR DESCRIPTION
**Pull request by aljesusg**
### Describe the change

Refactors the Kiali AI assistant architecture to support real-time token streaming (Server-Sent Events) and centralizes the conversation ID tracking directly in the backend. These changes significantly improve user experience by providing progressive text feedback and robust execution of AI tools.

### Tasks

- [X] Generate AI Conversation ID in the Backend
- [X] Cleanup: Removed obsolete API endpoints and handlers (ChatConversations, DeleteChatConversations) that were previously managing chat sessions manually, delegating lifecycle management entirely to the server configuration.
- [X] Streaming (Server-Sent Events) for AI Providers
- [X] Architecture Overhaul: Refactored the generic AIProvider interface to rely on a non-blocking onChunk callback mechanism instead of returning a synchronous response payload.
- [X] Real-time SSE: Added full streaming support (StreamTokenData, StreamStartData, StreamEndData, StreamToolCallData) for all 3 supported providers: OpenAI, Anthropic, and Google Gemini.
- [X] Streaming Tool Executions: Adapted the ExecuteToolCallsInParallel process to emit streaming events for tool_call and tool_result in real-time, allowing the UI to reflect parallel API steps transparently. Fixed edge cases in Gemini's and Anthropic's strict tool resolution loops (e.g., handling UI-only tools without hanging the stream).
- [X] Complete Tool Schema Test Coverage (Anthropic)
- [X] Test Suite Stabilization: Re-wrote tests across execution_test.go, helpers_test.go, and handlers/ai_test.go to safely test the new chunk-based streaming behavior and ensure total compatibility and reliability.
- [X] Simplify context passed to LLM and cleaned in system Instruction.
- [X] Implemented Dynamic Route Parsing 
- [X] Refined Chat UI Rendering
- [X] Added more cypress tests
- [X] Fixed openai streaming parallel
- [X] Improved context passed
- [X] Added translations
- [X] Added Lightspeed support
- [X] Added docs for lightspeed setup dev environment

## Related with
https://github.com/kiali/kiali/pull/9226
https://github.com/kiali/kiali/pull/9347

Operator PR https://github.com/kiali/kiali-operator/pull/1037
Helm PR https://github.com/kiali/helm-charts/pull/407

### CODE REVIEW

**Comment by hhovsepy** [comment]
I did find one issue, probably a regression as it works in master:
Asking "what means mesh health?"
It provides partly informaton.
Asking something else "give me all namespaces list".
It provides the second part of previous query, then processes the  next query.
Screenshot From 2026-05-21 18-37-16
Screenshot From 2026-05-21 18-37-37

**Comment by aljesusg** [comment]
> I did find one issue, probably a regression as it works in master: Asking "what means mesh health?" It provides partly informaton. Asking something else "give me all namespaces list". It provides the second part of previous query, then processes the next query. Screenshot From 2026-05-21 18-37-16 Screenshot From 2026-05-21 18-37-37

I'm unable to reproduce this. Which LLM and specific model version did you use?

Screenshot From 2026-05-25 08-49-59

File: ai/providers/helpers.go [hunk 1]
```diff
@@ -70,135 +175,71 @@ func CleanConversation(conversation *[]types.ConversationMessage) {
...
+			ptr.Conversation = append([]types.ConversationMessage(nil), storedConversation.Conversation...)
+			ptr.LastAccessed = storedConversation.LastAccessed
+			ptr.EstimatedMB = storedConversation.EstimatedMB
 		} else {
 			log.Debugf("Creating new conversation for conversation ID: %s", req.ConversationID)
-			// Create a new Conversation struct for storage later
-			ptr = &types.Conversation{}
+		}
+		if initializeConversation != nil {
+			initializeConversation(ptr, req.Query)
```
Comment (josunect) [inline ai/providers/helpers.go]: This should be called when the aiStore is disabled as well, shouldn't?

File: ai/providers/lightspeed/lightspeed_provider.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+type LightSpeedProvider struct {
+	client client.Client
+}
+
+// NewLightSpeedProvider creates a LightSpeed provider using only the
+// provider-level endpoint. LightSpeed has no models or API key; authentication
+// is handled per-request via the Kiali user's Kubernetes bearer token.
+func NewLightSpeedProvider(conf *config.Config, provider *config.ProviderConfig, model *config.AIModel) (*LightSpeedProvider, error) {
+	return &LightSpeedProvider{
+		client: *client.New(provider.Endpoint, client.WithInsecureSkipTLS(true)),
```
Comment (josunect) [inline ai/providers/lightspeed/lightspeed_provider.go:17]: Should this allow to include the certificates in the Kiali config? So it doesn't need to be always insecure

File: frontend/src/components/ChatBot/PageContext.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        break;
+      case 'istio':
+        context = 'User is seeing the istio objects list';
+        break;
+      default:
+        context = `User is seeing the ${kind} page`;
+    }
+
+    if (namespace) {
+      context += ` for namespaces: ${namespace}`;
```
Comment (josunect) [inline frontend/src/components/ChatBot/PageContext.tsx:36]: Should this be using name? As in `UseLocationContext` for kind namespace, the namespace is empty
```suggestion
      context += ` for namespaces: ${name}`;
```

File: frontend/src/components/ChatBot/Prompt.tsx [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+      conversation_id: conversationID,
+      query: pageContext ? `Context: ${pageContext}\n\n${query}` : query
+    };
+
+    const streamResponse = async (): Promise<void> => {
+      const controller = new AbortController();
+      setStreamController(controller);
+      const response = await API.postChatAI(selectedProvider, selectedModel, request, controller.signal);
+      // Problem with the Fetch
+      if (response.status !== 200) {
```
Comment (josunect) [inline frontend/src/components/ChatBot/Prompt.tsx:95]: Should this try to parse the error? 
Something like: 
```
let json: any | undefined;
  let text: string | undefined;
  const contentType = response.headers.get('content-type') || '';
  try {
    if (contentType.includes('application/json')) {
      json = await response.json();
    } else {
      text = await response.text();
    }
  } catch {
    // ignore parse errors and fall back below
  }
```

File: ai/providers/google/google_provider.go [hunk 5]
```diff
@@ -17,6 +17,10 @@ type GoogleAIProvider struct {
 	tracingEnabled bool
 }
 
+func (p *GoogleAIProvider) GetName() string {
+	return "Google"
+}
+
 func NewGoogleAIProvider(conf *config.Config, provider *config.ProviderConfig, model *config.AIModel) (*GoogleAIProvider, error) {
```
Comment (josunect) [inline ai/providers/google/google_provider.go:24]: Should this read the endpoint the same way as anthropic or openai? 

`opts.HTTPOptions.BaseURL = endpoint`

**Review by josunect — COMMENTED** [review]
Nice PR, @aljesusg ! 

Tested and looks good: 

Image

Image

Image

Image

Image

I've added some comments in the code. 

Some minor things, in the UI in the chat: 
The Kiali icon looks cut off: 

Image

Should this window have a max width and a title? 

Image

File: frontend/src/components/ChatBot/PageContext.tsx (hunk 3 above)
Comment (aljesusg) [reply frontend/src/components/ChatBot/PageContext.tsx:36]: No, This is only used when is filtered by namespaces.

File: frontend/src/components/ChatBot/Prompt.tsx (hunk 4 above)
Comment (aljesusg) [reply frontend/src/components/ChatBot/Prompt.tsx:95]: We are returning now events 

data: {json} So IMO it's not required due is ever a text stream

File: ai/providers/google/google_provider.go (hunk 5 above)
Comment (aljesusg) [reply ai/providers/google/google_provider.go:24]: This isn't needed for Google. Unlike OpenAI, where you can use an Azure bucket, Google works differently. I just used the default setting. In fact, it was already set up beforehand.

File: ai/providers/lightspeed/lightspeed_provider.go (hunk 2 above)
Comment (aljesusg) [reply ai/providers/lightspeed/lightspeed_provider.go:17]: Yes, but let's do that in a follow-up PR. I added a configuration to use it in-cluster, so I don't expect we'll need certificates here. Lightspeed is restricted in this regard since it doesn't have an exposed route. That's why I added documentation on how to configure a network policy to reach it.

File: ai/providers/helpers.go (hunk 1 above)
Comment (aljesusg) [reply ai/providers/helpers.go]: Well, we should probably keep conversations in memory at the very least. Otherwise, the chat will lose its context

File: frontend/src/components/ChatBot/PageContext.tsx (hunk 3 above)
Comment (josunect) [reply frontend/src/components/ChatBot/PageContext.tsx:36]: But then is going to pass some undefined value: 
image

And returning: 

```
data: {"data":{"conversation_id":"c6d528cd-b9dd-4df2-947e-0ad2250b2890"},"event":"start"}

data: {"data":{"input_tokens":0,"output_tokens":0,"actions":[{"title":"View Namespaces","kind":"navigation","payload":"/namespaces"}],"truncated":false},"event":"end"}
```

image

Comment (aljesusg) [reply frontend/src/components/ChatBot/PageContext.tsx:36]: A ok . I didn't test the new namespace page. Let me add the case

Comment (aljesusg) [reply frontend/src/components/ChatBot/PageContext.tsx:36]: Done

File: frontend/src/components/ChatBot/Prompt.tsx (hunk 4 above)
Comment (josunect) [reply frontend/src/components/ChatBot/Prompt.tsx:95]: Right, but the error is not handled as a stream event, so this processing would be just to pass these in the dispatch to show the error from the backend:
```
dispatch(
  ChatAIActions.setChatHistoryUpdateById({
    id: chatEntryID,
    entry: {
      error: getFetchErrorMessage({
        response,
        json,
        message: text
      }),
      who: 'ai',
      isStreaming: false,
      isTruncated: false
    }
  })
);
```

File: ai/providers/google/google_provider.go (hunk 5 above)
Comment (josunect) [reply ai/providers/google/google_provider.go:24]: In that case, if it is possible that the user setup the endpoint in the config, to throw a log message so the endpoint is not silently ignored?

File: ai/providers/helpers.go (hunk 1 above)
Comment (josunect) [reply ai/providers/helpers.go]: If the ai Store is disabled, now it returns: 
image
But, if `InitializeConversation()` is moved outside the `if aiStore.Enabled() `block, it works even when storage is disabled.

**Review by josunect — APPROVED** [review]
LGTM!

**Comment by hhovsepy** [comment]
> I'm unable to reproduce this. Which LLM and specific model version did you use?
> 

I was still getting that issue on 'gemini-2.5-flash', but that is a corner case, but anyway good PR @aljesusg thanks, otherwise worked fine for me.

**Comment by aljesusg** [comment]
> > I'm unable to reproduce this. Which LLM and specific model version did you use?
> 
> I was still getting that issue on 'gemini-2.5-flash', but that is a corner case, but anyway good PR @aljesusg thanks, otherwise worked fine for me.

Please open an issue. I'll try to reproduce it
